src/shared/cloudformation/cloudformation.ts

```diff
@@ -747,3 +759,133 @@ export namespace CloudFormation {
+
+export async function refreshSchemas(extensionContext: vscode.ExtensionContext) {
+    CFN_SCHEMA_PATH = path.join(extensionContext.globalStoragePath, 'cloudformation.schema.json')
+    SAM_SCHEMA_PATH = path.join(extensionContext.globalStoragePath, 'sam.schema.json')
+    let manifest: string | undefined
+    try {
+        const manifestFetcher = new HttpResourceFetcher(MANIFEST_URL, { showUrl: true })
+        manifest = await manifestFetcher.get()
+    } catch (e) {
+        getLogger().error(`Failed getting manifest at ${MANIFEST_URL}:`, e)
+
+        return
+    }
+
+    if (!manifest) {
+        getLogger().error(`Schema manifest at ${MANIFEST_URL} was undefined`)
+
+        return
+    }
+
+    try {
+        const details = await getManifestDetails(manifest)
+
+        await getRemoteOrCachedFile({
+            filepath: CFN_SCHEMA_PATH,
+            version: details.version,
+            url: details.cfnUrl,
+            cacheKey: 'cfnSchemaVersion',
+        })
+        await getRemoteOrCachedFile({
+            filepath: SAM_SCHEMA_PATH,
+            version: details.version,
+            url: details.samUrl,
+            cacheKey: 'samSchemaVersion',
+        })
+    } catch (e) {
+        getLogger().error('Could not get details from manifest:', (e as Error).message)
+    }
+}
+
+/**
+ * Maps a template file to a CFN or SAM schema.
+ * If present, removes association with the other type of schema.
+ * Does not modify other schemas not managed by AWS.
+ * @param path Template file path
+ * @param type Template type to use for filepath
+ */
+export async function updateYamlSchemasArray(path: string, type: 'cfn' | 'sam'): Promise<void> {
+    const config = vscode.workspace.getConfiguration('yaml')
+    const relPath = normalizeSeparator(getWorkspaceRelativePath(path) ?? path)
+    const schemas: { [key: string]: string | string[] } | undefined = config.get('schemas')
+    const writeTo = type === 'cfn' ? CFN_SCHEMA_PATH : SAM_SCHEMA_PATH
+    const deleteFrom = type === 'sam' ? CFN_SCHEMA_PATH : SAM_SCHEMA_PATH
+    let newWriteArr: string[] = []
+    let newDeleteArr: string[] = []
+
+    if (schemas) {
+        if (schemas[writeTo]) {
+            newWriteArr = Array.isArray(schemas[writeTo])
+                ? (schemas[writeTo] as string[])
+                : [schemas[writeTo] as string]
+            if (!newWriteArr.includes(relPath)) {
+                newWriteArr.push(relPath)
+            }
+        }
+        if (schemas[deleteFrom]) {
+            const temp = Array.isArray(schemas[deleteFrom])
+                ? (schemas[deleteFrom] as string[])
+                : [schemas[deleteFrom] as string]
+            newDeleteArr = temp.filter(val => val !== relPath)
+        }
+    }
+
+    config.update(
+        'schemas',
+        {
+            ...(schemas ? schemas : {}),
+            [writeTo]: newWriteArr,
+            [deleteFrom]: newDeleteArr,
+        },
+        vscode.ConfigurationTarget.Global
```

Comment:
One current flaw: this will take all `yaml.schemas` settings applicable to the workspace and write them to the global settings. I don't think there's a way around this; `vscode.workspace.getConfiguration` does not provide a way to pull a specific `vscode.ConfigurationTarget` (it can pull a `vscode.ConfigurationScope`, which is not the same).

Reply:
I'm thinking we need to investigate this more since applying this stuff to the global configuration is really undesirable. There probably is a way around it but it could be somewhat hacky.

Reply:
I received offline feedback from I think @justinmk3 to move this to global although I'm unsure why now (change was made here: https://github.com/aws/aws-toolkit-vscode/pull/1748/commits/d0751eb02030f230245175b710728f1a8165d647#diff-dcdb8f11716e3df28e8161a7fc741c5a4e8f2c8984ad5e971d0d1b3a49d0e40eL93 ) . I'm cool with moving this back to a workspace scope, although I'm thinking this may bring the reverse problem of including all global YAML configs into the workspace's config (but would leave out any other workspace-specific configs).

Reply:
Ah okay I see the problem now. These settings don't 'merge' through the scopes do they?

Reply:
When pulling, you choose a [configuration scope](https://code.visualstudio.com/api/references/vscode-api#ConfigurationScope) which I believe details all settings that are applicable to that scope, which includes any globals that affect the scope. When you save, you have to choose a [configuration target](https://code.visualstudio.com/api/references/vscode-api#ConfigurationTarget) which is the destination. So this means either global state is duplicated in the workspace configs, or all workspaces are forever held in the global config, including unaffiliated schemas used by the YAML extension.

Reply:
That being said, maybe a Uri type configuration scope could work here. Can investigate.

Reply:
My biggest concern is cluttering the user's settings. Attaching it within the scope of the document would be perfect if it works.